File: 2DVoxelGame/Src/World/Chunk/Chunk.cpp
#include "Chunk.h"

Chunk::Chunk(int chunk_x, int chunk_y) {
	this->chunk_x = chunk_x;
	this->chunk_y = chunk_y;
}

void Chunk::CleanUp() {
	mesh.CleanUp();
}

void Chunk::CreateMesh() {
	unsigned int current_vertex = 0;

	for (unsigned int x = 0; x < 40; x++) {
		for (unsigned int y = 0; y < 40; y++) {
			vertices.push_back(x);
			vertices.push_back(y);
			vertices.push_back(x);
			vertices.push_back(y + 1);
			vertices.push_back(x + 1);
			vertices.push_back(y);
			vertices.push_back(x + 1);
			vertices.push_back(y + 1);

			indices.push_back(current_vertex);
			indices.push_back(current_vertex + 1);
			indices.push_back(current_vertex + 2);
			indices.push_back(current_vertex + 2);
			indices.push_back(current_vertex + 1);
			indices.push_back(current_vertex + 3);
			current_vertex += 4;
		}
	}

	mesh.CreateMesh(vertices.data(), indices.data(), vertices.size(), indices.size());
}
void Chunk::Draw() {
	mesh.Draw();
}

File: 3DVoxelGame-Server-Client-Web/Client/Scripts/World/Chunk.js
class Chunk {
    x = 0;
    z = 0;

    mesh = null;

    constructor(x, z) {
        this.x = x;
        this.z = z;
        this.mesh = new BasicMesh();
        this.mesh.CreateMesh([0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 0], [0, 1, 2, 2, 1, 3]);
    }

    Update() {

    }

    Draw() {
        this.mesh.Draw();
    }
}